java/sqlcommenter-java/src/main/java/com/google/cloud/sqlcommenter/schibernate/SCHibernate.java

```diff
@@ -33,16 +28,24 @@
   public String inspect(String sql) {
     State state = State.Holder.get();
 
-    // Otherwise it is time to augment the SQL with information about the controller.
-    SpanContext spanContext = tracer.getCurrentSpan().getContext();
-    if (spanContext.isValid() && spanContext.getTraceOptions().isSampled()) {
-      // Since our goal at this point is NOT to mutate the threadlocal storage's state.
-      // yet we still need to carefully capture the Span information, we'll make
-      // a copy of the thread local storage but set the TraceId and SpanId.
-
-      // Finally replace this mvcState but it'll just be local to this function
-      // and not the final ThreadLocalStorage State.
-      state = State.newBuilder(state).withSpanContext(spanContext).build();
+    // Priority for OpenTelemetry commenting
+    io.opentelemetry.api.trace.SpanContext spanContextOT = io.opentelemetry.api.trace.Span.current().getSpanContext();
+
+    if (spanContextOT.isValid() && spanContextOT.isSampled()) {
```

Comment:
The OpenTelemetry context can be valid and not sampled, I don’t think the else clause should be invoked.

Reply:
A switch here perhaps would make things clearer:
```java
switch (spanContextOT.isValid()) {
case true:
     state = State....

default: // Let’s try using OpenCensus
      ...
}

Reply:
Resolved by moving checks for isSampled into nested flow